Add sample for use of ServiceBus SDK from browser (html)

ramya0820 opened the pull request:
For more context refer to #2550 

Since this will not be run on CI, it can be merged,
Ideally this can be merged after 2565 is resolved.

ramya-rao-a commented:
@bterlson The file was initially added inside a folder called "browser", but that couldnt be commited as that name is listed in gitignore :)

I do not foresee more samples for browser. Apart from the initialization of ServiceBusClient, I would expect users to refer to the samples in the `javascript` folder. Therefore, I suggested that we keep this sample as a stand-alone file

ramya-rao-a commented:
If we do indeed need more browser samples, then I would suggest the below structure
- browserSamples
- nodeSamples
   - javascript
   - typescript

ramya0820 commented:
> Can we simplify the sample by sending a single message? That would reduce the number of lines drastically and get user's attention to what is important
> 
> ```js
> await sender.send({ body: "Hello world!" });
> ```

I thought we specifically decided in meeting on 4/29 to reuse sendMessages.js sample just like in https://github.com/Azure/azure-sdk-for-js/pull/2555#issuecomment-487756742 and add in reference note to copy-paste other Javascript samples for trying them out.

Are we sure we want to revert to simplifying it to something different? I would recommend keeping samples consistent similar to other types as much as possible.

@ramya-rao-a @bterlson

bterlson commented:
I'm sympathetic to wanting simplification even though I find the list of scientists fun. Should we simplify the send message sample in a similar manner? Users can probably figure out how to adapt that trivially to send multiple messages in an array.

ramya-rao-a commented:
Apologies for the mixed signals you are getting here @ramya0820 
I'll try and explain my point of view here.

The main intention of this sample is to show how to import this library in browser sample and instantiate the Service Bus client. 

Below are the most important parts that should pop out from the sample without needing to look for it
- `<script src="../../browser/service-bus.js"></script>`
- `const SBC = Azure.Messaging.ServiceBus.ServiceBusClient;`
- `const ns = SBC.createFromConnectionString(connectionString);`

For completeness sake, I am ok with sending 1 message, receiving it and printing it to the console. But that is secondary.

The problem with the keeping this sample similar to the sendMessages.js is that it there are way too many lines of code distracting the user from the few lines that are important.

> I would recommend keeping samples consistent similar to other types as much as possible.

To achieve that level of parity, ideally, we should have multiple browser samples matching to each one of the samples in the javascript samples. But managing 3 versions of the samples (js, ts, browser) is an overhead.
Therefore, what I had suggested in one of the offline meetings was to take the `interactiveLogin.js` as an example. The main intention there was to show how to authenticate using interactive login, therefore it just does that, then adds pointers to other samples. Similarly, here the main intention is to show how to instantiate the Service Bus Client in a browser sample.

> Should we simplify the send message sample in a similar manner

Regarding whether we should update the sendMessages.js sample itself, yes that sample has scope for improvement, but is probably out of scope of this PR

ramya0820 commented:
> Apologies for the mixed signals you are getting here @ramya0820
> I'll try and explain my point of view here.
> 
> The main intention of this sample is to show how to import this library in browser sample and instantiate the Service Bus client.
> 
> Below are the most important parts that should pop out from the sample without needing to look for it
> 
> * `<script src="../../browser/service-bus.js"></script>`
> * `const SBC = Azure.Messaging.ServiceBus.ServiceBusClient;`
> * `const ns = SBC.createFromConnectionString(connectionString);`
> 
> For completeness sake, I am ok with sending 1 message, receiving it and printing it to the console. But that is secondary.
> 
> The problem with the keeping this sample similar to the sendMessages.js is that it there are way too many lines of code distracting the user from the few lines that are important.
> 
> > I would recommend keeping samples consistent similar to other types as much as possible.
> 
> To achieve that level of parity, ideally, we should have multiple browser samples matching to each one of the samples in the javascript samples. But managing 3 versions of the samples (js, ts, browser) is an overhead.
> Therefore, what I had suggested in one of the offline meetings was to take the `interactiveLogin.js` as an example. The main intention there was to show how to authenticate using interactive login, therefore it just does that, then adds pointers to other samples. Similarly, here the main intention is to show how to instantiate the Service Bus Client in a browser sample.
> 
> > Should we simplify the send message sample in a similar manner
> 
> Regarding whether we should update the sendMessages.js sample itself, yes that sample has scope for improvement, but is probably out of scope of this PR

Thanks for clarifying.
Please refer to latest commit, I've simplified and added comments to help.

@ramya-rao-a @bterlson

sy-huss commented:
Where is the <script src="../../browser/service-bus.js"></script>?

In the EventsHub sample, the code is complete. I cannot find the library service-bus.js without having to look through an unfamiliar repository.

HarshaNalluru commented:
Hey @sy-huss,

Publishing browser bundles has been discontinued.
If you'd like to use the library in the browser, follow one of the methods of bundling through webpack or rollup or parcel presented in [Bundling Azure SDK libraries for a browser](https://github.com/Azure/azure-sdk-for-js/blob/master/documentation/Bundling.md#bundling-azure-sdk-libraries-for-a-browser).
Please create a new issue and let us know in case you face any issues with the bundling and using the library in the browser.
